### PR TITLE
Mntor 1026

### PR DESCRIPTION
**Pull request by mansaj**
* Refactoring
* HIBP util
* csurf protection

### CODE REVIEW

**Review by toufali — APPROVED** [review]

File: src/package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "esbuild": "^0.15.10",
     "helmet": "^6.0.0",
     "svgo": "^2.8.0",
```
Comment (rhelmer) [inline src/package.json:39]: Already discussed in slack, going to see how difficult https://github.com/Psifi-Solutions/csrf-sync would be to add.

File: src/utils/hibp.js [hunk 2]
```diff
@@ -247,16 +233,35 @@ async function subscribeHash (sha1) {
   return await kAnonReq(path, options)
 }
 
+/**
+A range subscription can be deleted with the following request:
+DELETE /range/[hash prefix]
+
+There is one possible response code that can be returned:
+1. HTTP 200: Range subscription already exists
+
```
Comment (rhelmer) [inline src/utils/hibp.js:242]: This is valid JS but I think for JSDoc you want something more like:

```suggestion
/**
 * A range subscription can be deleted with the following request:
 * DELETE /range/[hash prefix]
 *
 * There is one possible response code that can be returned:
 * 1. HTTP 200: Range subscription already exists
 *
```

**Review by rhelmer — COMMENTED** [review]
This seems to work as advertised, just a minor style nit, and a note about the alternative csrf library we discussed in Slack.

File: src/package.json (hunk 1 above)
Comment (mansaj) [reply src/package.json:40]: I think with csrf-sync, we will now have to manually generate token per session or per request. 
The backend will generate via the `generateToken` method and the frontend will have to take that and add it to the request header in order for the request to go through..

File: src/utils/hibp.js (hunk 2 above)
Comment (mansaj) [reply src/utils/hibp.js]: Fixed

File: src/package.json (hunk 1 above)
Comment (mansaj) [reply src/package.json:40]: @rhelmer with doublecsrf I believe the same flow is still needed as `csrf-sync`... As in every request (POST PUT PATCH DELETE etc) needs the token in the request header from `generateToken`.

Comment (mansaj) [reply src/package.json:40]: But I think this is what's intended.. When the frontend sends requests via POST etc, we will have to be mindful to set the header `x-csrf-token`

**Review by rhelmer — APPROVED** [review]

**Pull request merged**